README.md

```diff
@@ -97,10 +101,11 @@ After cmake generation you should have something that looks like the following o
 Use `make install` to install Alure library in `C:\Program Files (x86)` for it to be available on your system.  
 Otherwise simply run `make` to build the library and if the cmake option was chosen, all the examples. 
 Note if you use mingw (or mingw-w64, the name is the same for both) you may need to use  `mingw32-make.exe`
-instead of `make`, and make sure that file is located in your path.  Note you may need to run `make install` as admin
+instead of `make`, and make sure that file is located in your path.  Note you may need to run `make install` as admin.
 
-When building you may have many warnings, this is mostly due to the backwards compatibility features included to have C++17 functionality work for C++11/14 compilers.
-As long as you the build was successful you should be fine.  If you run an example and nothing shows up, make sure you've added the bin (not just the library/lib) directory to the path.
+When building you may have many warnings, this is mostly due to the backwards compatibility features included to have 
```

Comment:
Out of curiosity, what warnings? GCC 4.9 seems to be the oldest version that works correctly and Travis builds it without any warnings (4.8 fails to build Alure because it has some bugs with its C++11 implementation). MSVC 2015 only warns about some casts to bool being a potential performance issue, which is unrelated to any compatibility code.

Reply:
I get a truck load of warnings, and actually a lot of them are deprecation issues. and unused I get a few odd ones thgough.  note I just cleaned and ran this on the branch I've been using for this, so the only thing that changed is the README.md:

	"...\cmake.exe" --build "...\alure\cmake-build-debug" --target all -- -j 2
	[  2%] Building CXX object CMakeFiles/alure2.dir/src/devicemanager.cpp.obj
	[  4%] Building CXX object CMakeFiles/alure2_s.dir/src/devicemanager.cpp.obj
	[  6%] Building CXX object CMakeFiles/alure2.dir/src/device.cpp.obj
	[  9%] Building CXX object CMakeFiles/alure2_s.dir/src/device.cpp.obj
	[ 11%] Building CXX object CMakeFiles/alure2.dir/src/context.cpp.obj
	[ 13%] Building CXX object CMakeFiles/alure2_s.dir/src/context.cpp.obj
	[ 15%] Building CXX object CMakeFiles/alure2_s.dir/src/buffer.cpp.obj
	[ 18%] Building CXX object CMakeFiles/alure2.dir/src/buffer.cpp.obj
	In file included from ...\alure\src\buffer.cpp:9:0:
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::BasicStringView<char>, alure::ChannelConfig, alure::SampleType, unsigned int, alure::ArrayView<signed char>) noexcept; Args = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, alure::ChannelConfig&, alure::SampleType&, unsigned int&, std::vector<signed char, std::allocator<signed char> >&}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
		 void send(R MessageHandler::* func, Args&&... args)
			  ^~~~
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::Source) noexcept; Args = {alure::SourceImpl*&}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
	In file included from ...\alure\src\buffer.cpp:9:0:
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::BasicStringView<char>, alure::ChannelConfig, alure::SampleType, unsigned int, alure::ArrayView<signed char>) noexcept; Args = {const std::__cxx11::basic_string<char, std::char_traits<char>, std::allocator<char> >&, alure::ChannelConfig&, alure::SampleType&, unsigned int&, std::vector<signed char, std::allocator<signed char> >&}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
		 void send(R MessageHandler::* func, Args&&... args)
			  ^~~~
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::Source) noexcept; Args = {alure::SourceImpl*&}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
	[ 20%] Building CXX object CMakeFiles/alure2_s.dir/src/source.cpp.obj
	[ 22%] Building CXX object CMakeFiles/alure2.dir/src/source.cpp.obj
	In file included from ...\alure\src\source.cpp:12:0:
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::Source) noexcept; Args = {alure::Source}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
		 void send(R MessageHandler::* func, Args&&... args)
			  ^~~~
	In file included from ...\alure\src\source.cpp:12:0:
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::Source) noexcept; Args = {alure::Source}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
		 void send(R MessageHandler::* func, Args&&... args)
			  ^~~~
	[ 25%] Building CXX object CMakeFiles/alure2_s.dir/src/sourcegroup.cpp.obj
	[ 27%] Building CXX object CMakeFiles/alure2.dir/src/sourcegroup.cpp.obj
	In file included from ...\alure\src\sourcegroup.cpp:7:0:
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::Source) noexcept; Args = {alure::SourceImpl*&}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
		 void send(R MessageHandler::* func, Args&&... args)
			  ^~~~
	In file included from ...\alure\src\sourcegroup.cpp:7:0:
	...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::Source) noexcept; Args = {alure::SourceImpl*&}]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]
		 void send(R MessageHandler::* func, Args&&... args)
			  ^~~~
	[ 29%] Building CXX object CMakeFiles/alure2_s.dir/src/auxeffectslot.cpp.obj
	[ 31%] Building CXX object CMakeFiles/alure2.dir/src/auxeffectslot.cpp.obj
	[ 34%] Building CXX object CMakeFiles/alure2_s.dir/src/effect.cpp.obj
	[ 36%] Building CXX object CMakeFiles/alure2.dir/src/effect.cpp.obj
	[ 38%] Building CXX object CMakeFiles/alure2_s.dir/src/decoders/wave.cpp.obj
	[ 40%] Building CXX object CMakeFiles/alure2.dir/src/decoders/wave.cpp.obj
	[ 43%] Building CXX object CMakeFiles/alure2_s.dir/src/decoders/vorbisfile.cpp.obj
	[ 45%] Building CXX object CMakeFiles/alure2.dir/src/decoders/vorbisfile.cpp.obj
	In file included from ...\alure\src\decoders\vorbisfile.cpp:8:0:
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:96:21: warning: 'OV_CALLBACKS_STREAMONLY_NOCLOSE' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_STREAMONLY_NOCLOSE = {
						 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:89:21: warning: 'OV_CALLBACKS_STREAMONLY' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_STREAMONLY = {
						 ^~~~~~~~~~~~~~~~~~~~~~~
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:82:21: warning: 'OV_CALLBACKS_NOCLOSE' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_NOCLOSE = {
						 ^~~~~~~~~~~~~~~~~~~~
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:75:21: warning: 'OV_CALLBACKS_DEFAULT' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_DEFAULT = {
						 ^~~~~~~~~~~~~~~~~~~~
	In file included from ...\alure\src\decoders\vorbisfile.cpp:8:0:
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:96:21: warning: 'OV_CALLBACKS_STREAMONLY_NOCLOSE' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_STREAMONLY_NOCLOSE = {
						 ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:89:21: warning: 'OV_CALLBACKS_STREAMONLY' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_STREAMONLY = {
						 ^~~~~~~~~~~~~~~~~~~~~~~
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:82:21: warning: 'OV_CALLBACKS_NOCLOSE' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_NOCLOSE = {
						 ^~~~~~~~~~~~~~~~~~~~
	C:/msys64/mingw64/include/vorbis/vorbisfile.h:75:21: warning: 'OV_CALLBACKS_DEFAULT' defined but not used [-Wunused-variable]
	 static ov_callbacks OV_CALLBACKS_DEFAULT = {
						 ^~~~~~~~~~~~~~~~~~~~
	[ 47%] Building CXX object CMakeFiles/alure2_s.dir/src/decoders/flac.cpp.obj
	[ 50%] Building CXX object CMakeFiles/alure2.dir/src/decoders/flac.cpp.obj
	[ 52%] Building CXX object CMakeFiles/alure2_s.dir/src/decoders/opusfile.cpp.obj
	[ 54%] Building CXX object CMakeFiles/alure2.dir/src/decoders/opusfile.cpp.obj
	[ 56%] Building CXX object CMakeFiles/alure2_s.dir/src/decoders/sndfile.cpp.obj
	[ 59%] Building CXX object CMakeFiles/alure2.dir/src/decoders/sndfile.cpp.obj
	[ 61%] Building CXX object CMakeFiles/alure2_s.dir/src/decoders/mpg123.cpp.obj
	[ 63%] Building CXX object CMakeFiles/alure2.dir/src/decoders/mpg123.cpp.obj
	[ 65%] Linking CXX static library libalure2_s.a
	[ 68%] Linking CXX shared library libalure2.dll
	[ 68%] Built target alure2_s
	[ 68%] Built target alure2
	[ 70%] Building CXX object CMakeFiles/alure-enum.dir/examples/alure-enum.cpp.obj
	[ 72%] Building CXX object CMakeFiles/alure-play.dir/examples/alure-play.cpp.obj
	[ 75%] Linking CXX executable alure-enum.exe
	[ 77%] Linking CXX executable alure-play.exe
	[ 77%] Built target alure-enum
	[ 77%] Built target alure-play
	[ 79%] Building CXX object CMakeFiles/alure-dumb.dir/examples/alure-dumb.cpp.obj
	In file included from ...\alure\examples\alure-dumb.cpp:13:0:
	C:/msys64/mingw64/include/dumb.h:41:19: warning: invalid suffix on literal; C++11 requires a space between literal and string macro [-Wliteral-suffix]
	 #define DUMB_NAME "DUMB v"DUMB_VERSION_STR
					   ^
	C:/msys64/mingw64/include/dumb.h:53:25: warning: invalid suffix on literal; C++11 requires a space between literal and string macro [-Wliteral-suffix]
	 #define DUMB_MONTH_STR2 "0"DUMB_MONTH_STR1
							 ^
	C:/msys64/mingw64/include/dumb.h:59:23: warning: invalid suffix on literal; C++11 requires a space between literal and string macro [-Wliteral-suffix]
	 #define DUMB_DAY_STR2 "0"DUMB_DAY_STR1
						   ^
	C:/msys64/mingw64/include/dumb.h:71:36: warning: invalid suffix on literal; C++11 requires a space between literal and string macro [-Wliteral-suffix]
	 #define DUMB_DATE_STR DUMB_DAY_STR1"."DUMB_MONTH_STR1"."DUMB_YEAR_STR4
										^
	C:/msys64/mingw64/include/dumb.h:71:54: warning: invalid suffix on literal; C++11 requires a space between literal and string macro [-Wliteral-suffix]
	 #define DUMB_DATE_STR DUMB_DAY_STR1"."DUMB_MONTH_STR1"."DUMB_YEAR_STR4
														  ^
	[ 81%] Building CXX object CMakeFiles/alure-hrtf.dir/examples/alure-hrtf.cpp.obj
	[ 84%] Linking CXX executable alure-dumb.exe
	[ 86%] Linking CXX executable alure-hrtf.exe
	[ 86%] Built target alure-dumb
	[ 86%] Built target alure-hrtf
	[ 88%] Building CXX object CMakeFiles/alure-stream.dir/examples/alure-stream.cpp.obj
	[ 90%] Building CXX object CMakeFiles/alure-reverb.dir/examples/alure-reverb.cpp.obj
	[ 93%] Linking CXX executable alure-stream.exe
	[ 93%] Built target alure-stream
	[ 95%] Linking CXX executable alure-reverb.exe
	[ 97%] Building CXX object CMakeFiles/alure-physfs.dir/examples/alure-physfs.cpp.obj
	[ 97%] Built target alure-reverb
	...\alure\examples\alure-physfs.cpp: In member function 'virtual std::basic_streambuf<char>::int_type {anonymous}::PhysFSBuf::underflow()':
	...\alure\examples\alure-physfs.cpp:33:13: warning: 'PHYSFS_sint64 PHYSFS_read(PHYSFS_File*, void*, PHYSFS_uint32, PHYSFS_uint32)' is deprecated [-Wdeprecated-declarations]
				 );
				 ^
	In file included from ...\alure\examples\alure-physfs.cpp:14:0:
	C:/msys64/mingw64/include/physfs.h:1326:27: note: declared here
	 PHYSFS_DECL PHYSFS_sint64 PHYSFS_read(PHYSFS_File *handle,
							   ^~~~~~~~~~~
	...\alure\examples\alure-physfs.cpp: In member function 'virtual std::basic_streambuf<char>::pos_type {anonymous}::PhysFSBuf::seekoff(std::basic_streambuf<char>::off_type, std::ios_base::seekdir, std::ios_base::openmode)':
	...\alure\examples\alure-physfs.cpp:91:52: warning: 'PHYSFS_sint64 PHYSFS_read(PHYSFS_File*, void*, PHYSFS_uint32, PHYSFS_uint32)' is deprecated [-Wdeprecated-declarations]
				 PHYSFS_read(mFile, mBuffer.data(), 1, 1);
														^
	In file included from ...\alure\examples\alure-physfs.cpp:14:0:
	C:/msys64/mingw64/include/physfs.h:1326:27: note: declared here
	 PHYSFS_DECL PHYSFS_sint64 PHYSFS_read(PHYSFS_File *handle,
							   ^~~~~~~~~~~
	...\alure\examples\alure-physfs.cpp: In static member function 'static void {anonymous}::FileFactory::ListDirectory(std::__cxx11::string&&)':
	...\alure\examples\alure-physfs.cpp:190:47: warning: 'int PHYSFS_isDirectory(const char*)' is deprecated [-Wdeprecated-declarations]
				 if(PHYSFS_isDirectory(file.c_str()))
												   ^
	In file included from ...\alure\examples\alure-physfs.cpp:14:0:
	C:/msys64/mingw64/include/physfs.h:1152:17: note: declared here
	 PHYSFS_DECL int PHYSFS_isDirectory(const char *fname) PHYSFS_DEPRECATED;
					 ^~~~~~~~~~~~~~~~~~
	[100%] Linking CXX executable alure-physfs.exe
	[100%] Built target alure-physfs

Reply:
> ```...\alure\src\context.h:277:10: warning: mangled name for 'void alure::ContextImpl::send(R alure::MessageHandler::*, Args&& ...) [with R = void(alure::BasicStringView<char>, alure::ChannelConfig, alure::SampleType, unsigned int, alure::ArrayView<signed char>) noexcept; Args = ...]' will change in C++17 because the exception specification is part of a function type [-Wnoexcept-type]```

Hmm, that's interesting that it warns about that. It's not related to the compatibility code though, that's just warning that when compiling as C++17 the mangled name for the ```alure::ContextImpl::send``` template function will be different when given functions marked ```noexcept```(it becomes part of the function type in C++17, when it wasn't in prior standards). It's also not an issue since that function is internal to the library, so code using it should be recompiled along with it.

> ```warning: 'OV_CALLBACKS_DEFAULT' defined but not used [-Wunused-variable]```

Odd that it warns about that for you, but not me or the Travis build. Seems there's a ```OV_EXCLUDE_STATIC_CALLBACKS``` macro that should be defined to avoid defining those static variables (would be better if the header could've marked them with a ```maybe_unused``` kind of attribute to suppress it).

> ```warning: 'PHYSFS_sint64 PHYSFS_read(PHYSFS_File*, void*, PHYSFS_uint32, PHYSFS_uint32)' is deprecated [-Wdeprecated-declarations]```
> ```warning: 'int PHYSFS_isDirectory(const char*)' is deprecated [-Wdeprecated-declarations]```

I guess a newer version of PhysFS deprecates those functions. 3.0 was released not too long ago, but most distros still have 2.0 in their stable repo.

Reply:
OK, i'll remove that part about warnings alltogether then.